rskj-core/src/test/java/co/rsk/vm/opcode/TransientStorageDslTest.java

```diff
@@ -341,6 +341,96 @@ void testDynamicExecutionStaticCallSubcallCantUseTstore() throws FileNotFoundExc
+        assertTransactionReceiptWithStatus(world, txTstorageDynamicReentrancyContextContract, "b01", true);
+
+        String txTestReentrantContextRevert = "txTestReentrantContextRevert";
+        TransactionReceipt txReceipt = assertTransactionReceiptWithStatus(world, txTestReentrantContextRevert, "b02", true);
+        Assertions.assertEquals(3, TransactionReceiptUtil.getEventCount(txReceipt, "OK",  null));
+
+        String txTestReentrantContextInvalid = "txTestReentrantContextInvalid";
+        assertTransactionReceiptWithStatus(world, txTestReentrantContextInvalid, "b03", false);
+    }
+
+    @Test
+    void testDynamicReentrancyContextsRevertOrInvalidUndoesAll() throws FileNotFoundException, DslProcessorException {
+        DslParser parser = DslParser.fromResource("dsl/transaction_storage_rskip446/dynamic_reentrancy_context_revert_or_invalid_undoes_all.txt");
+        World world = new World();
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        String txTstorageDynamicReentrancyContextContract = "txTstorageDynamicReentrancyContextContract";
+        assertTransactionReceiptWithStatus(world, txTstorageDynamicReentrancyContextContract, "b01", true);
+
+        String txTestReentrantContextRevert = "txTestReentrantContextRevert";
+        TransactionReceipt txReceipt = assertTransactionReceiptWithStatus(world, txTestReentrantContextRevert, "b02", true);
+        Assertions.assertEquals(5, TransactionReceiptUtil.getEventCount(txReceipt, "OK",  null));
+
+        String txTestReentrantContextInvalid = "txTestReentrantContextInvalid";
+        assertTransactionReceiptWithStatus(world, txTestReentrantContextInvalid, "b03", false);
+    }
+
+    @Test
+    void testReentrancyContextsTstoreAfterReentrantCall() throws FileNotFoundException, DslProcessorException {
+        DslParser parser = DslParser.fromResource("dsl/transaction_storage_rskip446/reentrancy_context_tstore_after_reentrant_call.txt");
+        World world = new World();
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        String txTstorageReentrancyContextTestContract = "txTstorageReentrancyContextTestContract";
+        assertTransactionReceiptWithStatus(world, txTstorageReentrancyContextTestContract, "b01", true);
+
+        String txTstoreInReentrantCall = "txTstoreInReentrantCall";
+        TransactionReceipt txReceipt = assertTransactionReceiptWithStatus(world, txTstoreInReentrantCall, "b02", true);
+
+        String txCheckValuesStoredInTstorage = "txCheckValuesStoredInTstorage";
+        txReceipt = assertTransactionReceiptWithStatus(world, txCheckValuesStoredInTstorage, "b03", true);
+        Assertions.assertEquals(3, TransactionReceiptUtil.getEventCount(txReceipt, "OK",  null));
+    }
+
+    @Test
+    void testReentrancyContextsTloadAfterReentrantTstore() throws FileNotFoundException, DslProcessorException {
```

Comment:
Same coment that in the previous test.

Reply:
Maybe add something like "ExecutesAsExpected" or "DoAsExpected" at the end of the names.

Reply:
Same comment from above, this scenario is validating a tload after a reentrant call with tstore. Do you think I should expand it more?